### PR TITLE
ceph.spec.in: Forward port suse build deps and recomends

### PR DESCRIPTION
**Pull request by osynge**
SUSE spec file got very out of sync with upstream master. This patch
helps resolve teh build depednancies from suse ceph.

Signed-off-by: Owen Synge <osynge@suse.com>

### CODE REVIEW

**Comment by osynge** [comment]
ops!

Got to correct spoelling error in commit message.

**Comment by osynge** [comment]
Should also state this is tested on fedora 21 / 22.

**Comment by osynge** [comment]
Spelling error in commit message fixed.

**Comment by osynge** [comment]
Please note that if this and https://github.com/ceph/ceph/pull/4793 are merged OBS can build upstream master packages. (It fails due to other rpm lint warnings but hey I can fix that)

**Comment by tchaikov** [comment]
just one concern, i see you are removing some of the build dependencies from distros where none of suse_version, rhel or fedora macros is defined.

what is our policy of supporting these unknown distros? shall we print out error messages or simply bail out? or we loose the build/runtime requirement, because we are not sure about the package names.

**Comment by osynge** [comment]
I am happy to follow what ever policy is chosen.

My preference on this subject of dependency and build dependency rpms:

Make sure autotools reports any missing dep to provide a clear error.

When rpm's don’t have consistent names between SUSE and Redhat derivatives (or any other rpm based distro) we don’t use "else" as we simply don’t know know what an unnamed distribution would use.

My rationality is that if the error is clear, then people can easily fix for said unknown distribution rather than assume any unnamed distribution is going to be like some other distribution.

**Comment by ktdreyer** [comment]
I don't know of any RPM-based distros apart from the SUSE family and Red Hat family, so I think we're fine in that regard.

File: ceph.spec.in [hunk 1]
```diff
@@ -123,8 +127,8 @@ BuildRequires:	keyutils-libs-devel
 BuildRequires:	libatomic_ops-devel
 Requires:	gdisk
 Requires(post):	chkconfig
-Requires(preun):chkconfig
-Requires(preun):initscripts
+Requires(preun):	chkconfig
+Requires(preun):	initscripts
```
Comment (ktdreyer) [inline ceph.spec.in]: In the future, this is one of those whitespace changes I'd prefer to be broken out into a separate commit... but I'll let it slide for this PR :)

File: ceph.spec.in [hunk 2]
```diff
@@ -103,6 +98,15 @@ BuildRequires:	sharutils
...
 %if 0%{defined suse_version}
+BuildRequires:	python-sphinx
+BuildRequires:	net-tools
+BuildRequires:	xmlstarlet
+BuildRequires:	yasm
+BuildRequires:	libbz2-devel
+BuildRequires:	python-sphinx
+BuildRequires:	net-tools
+BuildRequires:	sharutils
+BuildRequires:	git
```
Comment (ktdreyer) [inline ceph.spec.in]: With this change, there's a lot of duplication between what SUSE BuildRequires and what RHEL BuildRequires, so `xmlstarlet`, `yasm`, etc are listed twice in `ceph.spec.in` now. We could probably do another PR to merge these two lists.

**Comment by ktdreyer** [comment]
This change looks fine to me. merging.
